### PR TITLE
Add execute_ and has_? workflow methods

### PR DESCRIPTION
**Pull request by rwsu**
Adding these methods provide a standard set of interfaces to
interact with the Workflow service. It will also allow us to remove
some duplicate code in Host.

### CODE REVIEW

File: lib/manageiq/providers/openstack/legacy/openstack_handle/workflow_delegate.rb [hunk 1]
```diff
@@ -12,5 +12,62 @@ def initialize(dobj, os_handle, name)
...
+      [workflow_execution_id, workflow_state, response]
+    end
+
+    def has_workflow?(workflow_name)
+      result = false
+      @os_handle.service_for_each_accessible_tenant(SERVICE_NAME) do |svc|
+        begin
+          response = svc.get_workflow(workflow_name)
+          result = true if response.status == 200
+        rescue => err
```
Comment (chessbyte) [inline lib/manageiq/providers/openstack/legacy/openstack_handle/workflow_delegate.rb]: Would this be better as `return true if response.status == 200`?

Comment (rwsu) [inline lib/manageiq/providers/openstack/legacy/openstack_handle/workflow_delegate.rb]: Sure :)

**Comment by tzumainn** [comment]
Would it make sense to include the ManageIQ task queue as part of the workflow execution?  I can't imagine we'd ever execute a workflow outside of a task.  Or do you think that should remain separate?

**Comment by rwsu** [comment]
@tzumainn, I would lean towards not including the task queue as part of workflow execution. After successful execution, we currently kick off an ems refresh. Having the task queue as part of workflow execution would mean that we would need to pass a reference to the ems as a parameter to the workflow execution so that it kicks off the refresh if successful. Doing so makes the workflow delegate do more than it should. 

And we can't queue the ems refresh at the stack level because once the execution is queued, the stack doesn't have any visibility to the execution result,

**Comment by tzumainn** [comment]
@rwsu Fair enough!  The code looks good to me; I'll merge once tests pass.

**Comment by aufi** [comment]
Looks good to me!

**Pull request merged**